docs/proposals/images/cluster-class/update.plantuml

```diff
@@ -0,0 +1,45 @@
+@startuml
+title Figure 2. Cluster update with ClusterClass process
+actor User
+
+' -- GROUPS START ---
+
+box #LightGreen
+participant "API Server"
+end box
+
+box #LightBlue
+participant "Cluster Controller"
+end box
+
+' -- GROUPS END ---
+
+User -> "API Server" : Update Cluster object with\n""Cluster.Spec.Class""
+"API Server" --> "Cluster Controller": Existing Cluster
+opt Required only if Cluster.Spec.Class is set
+  "Cluster Controller" -> "Cluster Controller": Fetches the KCP set in\n Cluster.Spec.ControlPlaneRef
+  opt Update the ControlPlaneRef
+    "Cluster Controller" -> "Cluster Controller": Updates the number of replicas
+    "Cluster Controller" -> "Cluster Controller": Updates the k8s version
+    "Cluster Controller" -> "API Server": Updates KCP
```

Comment:
nit: let's make sure we don't assume KCP anywhere as _the_ control plane since there might be other control plane provider implementations in the future, and KCP will just be one of them